bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py

```diff
@@ -120,16 +144,16 @@ def _call_rdf_operation(distrdf_node: Node, previous_rdf_node: Any, range_id: in
     future_results = []
 
     rdf_operation = getattr(previous_rdf_node, distrdf_node.operation.name)
-    _make_op_lazy_if_needed(distrdf_node.operation, range_id)
-    pyroot_node = rdf_operation(*distrdf_node.operation.args, **distrdf_node.operation.kwargs)
+    in_task_op = _create_lazy_op_if_needed(distrdf_node.operation, range_id)
```

Comment:
So two tasks can invoke `_call_rdf_operation` on the same `distrdf_node` object? How does this happen, I thought every task generates its own graph?

Reply:
Every task generates its own RDataFrame C++ graph, but the DistRDF Python graph is a single object that gets serialized/deserialized. On a single machine, a single Python process can receive two(or more) tasks. When the first task starts, it deserializes the distrdf graph, modifies *in-place* the `operation` object, then it creates the RDF C++ calls, sends everything to the mapper function that executes them. When the second task starts, it gets the same distrdf graph objects,  but at this point their `operation` attributes were modified by the previous task, thus leading to the errors described in the PR

Reply:
I see, thanks!